capture/src/sinks/kafka.rs

```diff
@@ -137,22 +139,27 @@ impl KafkaSink {
         self.producer.flush(Duration::new(30, 0))
     }
 
-    async fn kafka_send(
-        producer: FutureProducer<KafkaContext>,
-        topic: String,
-        event: ProcessedEvent,
-        limited: bool,
-    ) -> Result<DeliveryFuture, CaptureError> {
+    async fn kafka_send(&self, event: ProcessedEvent) -> Result<DeliveryFuture, CaptureError> {
         let payload = serde_json::to_string(&event).map_err(|e| {
             error!("failed to serialize event: {}", e);
             CaptureError::NonRetryableSinkError
         })?;
 
-        let key = event.key();
-        let partition_key = if limited { None } else { Some(key.as_str()) };
+        let event_key = event.key();
+        let (topic, partition_key): (&str, Option<&str>) = match &event.data_type {
+            DataType::AnalyticsHistorical => (&self.historical_topic, Some(event_key.as_str())), // We never trigger overflow on historical events
```

Comment:
I wish we could get rid of locality in historical too.

Reply:
Blocked by person processing idempotence, but to be kept in mind. Part of the solution for person processing might be in-memory caches that would benefit from locality.
If we get more instances of historical being overwhelmed, we could implement a very-high threshold overflow detection here?

Reply:
I'm particularly concerned since historical doesn't overflow, but we also do have a higher SLA.

I do understand that the solution is not here but in person processing. Hoping there won't be much trouble until we get there.